src/dotnet/commands/dotnet-compile-csc/Program.cs

```diff
@@ -84,6 +84,7 @@ public static int Run(string[] args)
 
                 // Execute CSC!
                 var result = RunCsc(new string[] { $"-noconfig", "@" + $"{rsp}" })
+                    .WorkingDirectory(Directory.GetCurrentDirectory())
```

Comment:
Currently I only see BuiltInCommand implementing WorkingDirectory, but RunCsc returns a Command. Shouldn't you add WorkingDirectory to ICommand, and then have Command and BuiltInCommand implement it? 
Maybe have the common code in an AbstractCommand